src/entry/logEntry.store.ts

```diff
@@ -95,7 +69,95 @@ export class LogEntryStore {
       return result.rows;
     } catch (e) {
       console.log(e);
-      return LogEntryRequestError.serverError;
+      return RequestError.serverError;
+    }
+  }
+
+  async getIPs(
+    sessionID: UUID,
+    ipRequestData: IPRequestData,
+  ): Promise<string[] | RequestError> {
+    try {
+      let queryParams: any[] = [sessionID, ipRequestData.files];
+      let queryString = `
+      SELECT service_ip
+            FROM loggaroo.log_entry
+              WHERE session_id = $1
+                AND file_name = ANY($2)
+                `;
+
+      if (ipRequestData.filters) {
+        const queryData = this.applyFilters(
+          ipRequestData.filters,
+          queryString,
+          queryParams,
+        );
+
+        queryParams = queryData.queryParams;
+        queryString = queryData.queryString;
+      }
+
+      const query = {
+        text: queryString,
+        values: queryParams,
+      };
+
+      const result = await DatabaseService.getInstance()
+        .getClient()
+        .query<IP>(query);
+
+      const ipObjs = result.rows;
+
+      const ips: string[] = [];
+      for (const ip of ipObjs) {
+        ips.push(ip.ip);
+      }
+
+      return ips;
+    } catch (e) {
+      console.log(e);
+      return RequestError.serverError;
     }
   }
+
+  applyFilters(filters: Filters, queryString: string, queryParams: any[]) {
```

Comment:
This code modifies the original queryParams array I think. So clone the array before using it:
```suggestion
  applyFilters(filters: Filters, queryString: string, queryParams: any[]) {
      queryParams = [...queryParams];
```